### PR TITLE
E2 1802 fcm gcm

### PR DESCRIPTION
**Pull request by hrishileanplum**
Fixes #
The goal of this ticket is to remove GCM support from Leanplum Android sdk. As starting APRIL 11th, GCM will no longer be supported by google.

## Proposed Changes
https://leanplum.atlassian.net/browse/E2-1802

Cleanup the code, get rid of GCM and refactor to use the new correct FCM standard for fetching tokens.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ####################################################################
-#
+#
```
Comment (e7mac) [inline Makefile]: could we revert all the changes to this file?

File: settings.gradle [hunk 2]
```diff
@@ -2,6 +2,5 @@ include ':AndroidSDKCore'
 include ':AndroidSDKLocation'
 include ':AndroidSDKPush'
```
Comment (e7mac) [inline settings.gradle]: are we thinking of combining leanplum-push and leanplum-fcm in a future PR ?

File: Makefile (hunk 1 above)
Comment (hrishileanplum) [reply Makefile]: sure

Comment (hrishileanplum) [reply Makefile]: Actually now i realized. i had got merge conflicts with this file somehow. Thats the reason.

File: settings.gradle (hunk 2 above)
Comment (hrishileanplum) [reply settings.gradle]: Yes. In future for sure or in our new architecture . I tried doing it. But, there is lot more changes than i expected. So, not for this commit.

**Comment by milos1290** [comment]
Looking good, just one note, since we are modifying public API we need to bump major version.

**Pull request merged**